library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopVertex.java

```diff
@@ -63,26 +69,14 @@ public <V> VertexProperty<V> property(final String key) {
     @Override
     public <V> VertexProperty<V> property(final VertexProperty.Cardinality cardinality, final String key, final V value, final Object... keyValues) {
         if (isReadOnly()) {
-            throw new UnsupportedOperationException("Updates are not supported");
-        }
-
-        ElementHelper.legalPropertyKeyValueArray(keyValues);
-        ElementHelper.validateProperty(key, value);
-        final Optional<VertexProperty<V>> optionalVertexProperty = ElementHelper.stageVertexProperty(this, cardinality, key, value, keyValues);
-        if (optionalVertexProperty.isPresent()) {
-            return optionalVertexProperty.get();
-        }
-
-        final VertexProperty<V> vertexProperty = new GafferPopVertexProperty<>(this, key, value);
-
-        if (null == this.properties) {
-            this.properties = new HashMap<>();
+            throw new UnsupportedOperationException("Updates are not supported, Vertex is readonly");
         }
+        // Attach the property to this vertex before updating and re adding to the graph
+        VertexProperty<V> vertexProperty = propertyWithoutUpdate(cardinality, key, value, keyValues);
+        LOGGER.warn("Updating Vertex properties via aggregation");
 
-        final List<VertexProperty> list = this.properties.getOrDefault(key, new ArrayList<>());
-        list.add(vertexProperty);
-        this.properties.put(key, list);
-        ElementHelper.attachProperties(vertexProperty, keyValues);
+        // Re add to do a update via aggregation
+        this.graph().addVertex(this);
```

Comment:
Misuse of 'this'

Reply:
Technically not misuse more not required, the `graph()` method is in the parent class so might not have been obvious where it came from with out a `this`, I can remove though. I was mostly following the style the file already had where `this` was used to refer to anything that came from the parent class e.g. `properties` too.